eox_core/api/v1/tests/test_users.py

```diff
@@ -78,68 +160,216 @@ def test_update_role_not_allowed(self, get_edxapp_user, signup_source, _):
+        get_edxapp_user.return_value = self.user
+        signup_source.objects.filter.return_value.count.return_value = 1
+
+        response = self.client.patch(self.url, data=update_data, format="json")
+
+        self.assertRaises(ValidationError)
+        self.assertEqual(status.HTTP_400_BAD_REQUEST, response.status_code)
+        self.assertEqual(response.content, '{"detail":["You are not allowed to update address."]}'
+                         .encode())
+
+    @override_settings(EOX_CORE_USER_UPDATE_SAFE_FIELDS=["is_active", "password", "fullname"])
+    @patch_permissions
+    @patch('eox_core.api.v1.serializers.UserSignupSource')
+    @patch('eox_core.api.v1.views.get_edxapp_user')
+    @patch('eox_core.api.v1.views.EdxappUserReadOnlySerializer')
+    def test_update_non_safe_user_profile_fields(self, user_serializer, get_edxapp_user, signup_source, _):
+        """If a non safe UserProfile field wants to be updated, then the request must return a
+        400_BAD_REQUEST status code with the proper validation error message.
+        """
+        update_data = {
+            "email": "test",
+            "fullname": "test",
+            "country": "gr",
+        }
+        user_serializer.return_value.data = {}
+        get_edxapp_user.return_value = self.user
+        signup_source.objects.filter.return_value.count.return_value = 1
+
+        response = self.client.patch(self.url, data=update_data, format="json")
+
+        self.assertRaises(ValidationError)
+        self.assertEqual(status.HTTP_400_BAD_REQUEST, response.status_code)
+        self.assertEqual(response.content, '{"detail":["You are not allowed to update country."]}'
+                         .encode())
+
+    @patch_permissions
+    @patch('eox_core.api.v1.serializers.UserSignupSource')
+    @patch('eox_core.api.v1.views.get_edxapp_user')
+    @patch('eox_core.api.v1.views.EdxappUserReadOnlySerializer')
+    def test_fail_to_update_non_configured_extra_fields(self, user_serializer, get_edxapp_user, signup_source, _):
+        """Tests the WrittableEdxappUserSerializer. A field that has not been properly added
+        to the registration extra fields settings can't be updated.
+        """
+        update_data = {
+            "email": "test@example.com",
+            "fullname": "test",
+            "country": "ve",
+            "year_of_birth": 1997,
+            "favorite_food": "pizza",
+            "level_of_education": "m",
+        }
+        validated_data = {
+            "email": "test@example.com",
+            "fullname": "test",
+            "country": "ve",
+        }
+        user_serializer.return_value.data = {}
+        get_edxapp_user.return_value = self.user
+        signup_source.objects.filter.return_value.count.return_value = 1
+
+        serializer = WrittableEdxappUserSerializer(self.user, data=update_data, partial=True)
+        serializer.is_valid()
+
+        self.assertEqual(validated_data, serializer.data)
```

Comment:
how are you checking that can't be updated?

Reply:
@mariajgrimaldi  Basically what I'm testing here is that the WrittableEdxappUserSerializer updates its fields according to the site settings. The validated data that is being returned by the serializer only has the fields that are properly configured in the extra registration fields settings, if a field that is not there is sent in the request data, the serializer simply ignores it.
Maybe I should change the name of the test